net/src/lib.rs

```diff
@@ -0,0 +1,373 @@
+//! holochain_net is a library that defines an abstract networking layer for
+//! different network transports and implements a PeerStore for mapping and
+//! managing the topology of transport layers with regard to relay's e.g. for NAT
+
+#[macro_use]
+extern crate failure;
+
+use failure::Error;
+use std::sync::{Arc, Mutex};
+
+pub mod error;
+
+pub type SerializedAddress = Vec<u8>;
+pub type TransportAddress = Vec<u8>;
+pub type SerializedMessage = Vec<u8>;
+
+/// this closure type will get called when the send completes and the parameter will be the response message (or error)
+type SendResponseClosure = Box<FnMut(Result<SerializedMessage, Error>) -> Option<Error> + Send>;
+
+/// this closure type gets called when a new message arrives, you can respond with a Message or an error.
+type ReceiveClosure =
+    Box<FnMut(&SerializedAddress, &SerializedMessage) -> Result<SerializedMessage, Error> + Send>;
+
+pub trait Node {
+    fn get_address(&self) -> &SerializedAddress;
+    fn get_transport_address(&self) -> TransportAddress;
+}
+
+pub struct Handler {
```

Comment:
This wrapper seems unnecessary and to cause unneeded complexity

Reply:
Yes, I think you are right.  I ended up adding the wrapper in the phase when I was trying to get Transport to work as a trait object, and that matches how @lucksus wrapped up the closure in instance.rs.   I'll try simplifying back down...